### ISSUE TITLE
NVDA randomly freezing on Lenovo Thinkpad E14 5th gen

### ISSUE THREAD

**Issue by mayson199514** [nvaccess/nvda#15366]
### Steps to reproduce:
Crashes are too random to reliably reproduce.    
### Actual behavior:

NVDA randomly crashes under normal usage. 
### Expected behavior:

NVDA should not be frequently freezing under normal usage. 
### NVDA logs, crash dumps and other attachments:
refer to the attached log for details. 
#### NVDA installed/portable/running from source:
NVDA installed 
#### NVDA version:
NVDA 2023.2
#### Windows version:
Windows 11 Version 22H2 (OS Build 22621.2215) 
#### Other information about your system:
Refer to attached system information.
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Tried NVDA Alpha Snapshots, and they produced the same freezing behavior. 
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes 
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes
[nvda-old.log](https://github.com/nvaccess/nvda/files/12517826/nvda-old.log)
[system info.txt](https://github.com/nvaccess/nvda/files/12517830/system.info.txt)

**Comment by seanbudd** [comment]
This appears to be an issue with the add-on IBMTTS. However you say

>If NVDA add-ons are disabled, is your problem still occurring?
>>Yes

Can you confirm you've had a crash occur with all add-ons disabled? If so, can you provide a log of this occurring next time it happens?

**Comment by mayson199514** [comment]
Yes, it also happens with add-ons disable. hear is another log with espeak ng loaded. it doesn't matter what synthasisor I have loaded, it will still freez for a certain ammount of seconds. the most I seen it freeze and recover was 
[nvda-old.log](https://github.com/nvaccess/nvda/files/12551661/nvda-old.log)
29 seconds.

**Comment by mayson199514** [comment]
Hey Sean, its Mayson. I think my nvda freezing issue is partially associated with battery saver on windows 11. Thanks Sean, and will keep you posted on what else I observe. Aloha, Mayson. 

________________________________________

**Comment by mayson199514** [comment]
Hear is a new nvda log with addons disabled. I caught the freeze with debugging enabled. I hope this will Help!
[nvda-old september 8, 2023.log](https://github.com/nvaccess/nvda/files/12562931/nvda-old.september.8.2023.log)
 I thought it was related to battery saver, but apparently its not.

**Comment by seanbudd** [comment]
Can you confirm that this isn't a new issue i.e. that it occurred in NVDA versions prior to 2023.2?

This seems to be a known issue with NV Wave, which will soon be replaced by WASAPI in 2023.3.
For 2023.2, if you enable "Use WASAPI for audio output" in advanced preferences, this issue should go away.

**Comment by mayson199514** [comment]
Hello Sean, I can also confirm that this also happens on older nvda versions. Also, I also caught the freeze when running with wasapi on alpha snapshots. I'm working on getting another debug log with addons disabled to catch the freeze again so I'll be uploading that as soon as I can get the freeze to accure again with wasapi. Aloha, Mayson.

________________________________________

**Comment by mayson199514** [comment]
Hello sean, its Mayson again. I caught the freeze again this is not as long as the 29 second freeze that I offten get, but hopefully this time, the dibug log should help. I'm currently running nvda 2023.3 beta 1. it doesn't matter what application I am in. this freeze like I said happens randomly and is unreliably  reproducible. I've been running with addons disabled and debug logging enabled for the past week trying my hardest to catch the freeze, I caught it just now and I am using wasapi support that's enabled by default in 2023.3 beta 1. hopefully you guys can track this saveer bug down. this random freeze especially the 29 second freeze I offten get affects my work flow. thanks Sean and all, and look forward to a fix real soon. Aloha, Mayson.
[nvda september 17 2023.log](https://github.com/nvaccess/nvda/files/12642349/nvda.september.17.2023.log)

**Comment by mayson199514** [comment]
Hey Sean, hear is another log, this time using nvda alpha snapshots. Aloha, Mayson[nvda november 4, 2023.log](https://github.com/nvaccess/nvda/files/13257208/nvda.november.4.2023.log)

**Comment by yukie-nobuharu** [comment]
I will also confirm that the freezing also happens to me with addons enabled or disabled. my system info is attached in the file.  My nvda is installed. I also tried running nvda portable but, still the same. I'm using NVDA 2023.3. I will try to provide logs when it happens again.
[system.txt](https://github.com/nvaccess/nvda/files/13257686/system.txt)

**Comment by Adriani90** [comment]
@mayson199514 this log is not really useful because NVDA does not seem to crash or freeze here completely. Instead it seems it freezes briefly when you are using thunderbird.
The log is interupted at the end when you seem to save the log file, in the save as dialog. so in case NVDA was crashing completely, you should find a file called nvda_old.log in %temp%. Could you please upload that file to this issue?
1. Did you try to run the COM registration fixing tool after rebooting the computer?
2. Did you try to remove NVDA from your system, delete the user configuration in %appdata% and install NVDA again?
3. Did you try to run NVDA as administrator? Does then the freeze still occur?

**Comment by mayson199514** [comment]
[nvda-old november 19, 2023.log](https://github.com/nvaccess/nvda/files/13404667/nvda-old.november.19.2023.log)
 Hear is another log file. that showcases it freezing. I had to wait for a while for it to recover. after closing the new outlook that comes in windows 11

**Comment by mayson199514** [comment]
here is yet another log of the freezing accuring with all addons disabled this also happens with addons enabled too. 
[nvda-old october 30, 2023.log](https://github.com/nvaccess/nvda/files/13515537/nvda-old.october.30.2023.log)

**Comment by mayson199514** [comment]
sorry guys the log should say november 30, not october 30.

**Comment by Adriani90** [comment]
@mayson199514 you didn't answer my questions above. This cannot be further investigated without your help in this regard.

1. Did you try to run the COM registration fixing tool after rebooting the computer?
2. Did you try to remove NVDA from your system, delete the user configuration in %appdata% and install NVDA again?
3. Did you try to run NVDA as administrator? Does then the freeze still occur?

**Comment by Adriani90** [comment]
One additional question: does this freeze also occur with Narator? It might also be related toa driver that slows down your Laptop in some cases.

**Comment by mayson199514** [comment]
Yes, to all, and the freeze still happens.
________________________________

**Comment by mayson199514** [comment]
this issue does not happen with Narrator.
________________________________

**Comment by Adriani90** [comment]
@mayson199514 could you please test with NVDA 2024.1 Beta 10? Is this issue still occuring?

**Comment by seanbudd** [comment]
is this a freeze or a crash? a crash means the software exits unexpectedly. A freeze is recovered from without needing to start NVDA again.

**Comment by bhavyashah** [comment]
@mayson199514 I also have a Lenovo Thinkpad - different model though - on which I have experienced something similar to what you seem to be describing. When NVDA goes silent and has crashed, my experience is as follows:
* Tabbing or arrowing or speech-inducing NVDA commands (like NVDA+t) produce no sound.
* Pressing the restart NVDA command (mine is set to Ctrl+Alt+/) doesn't work either. 
* I enable Narrator (Ctrl+Windows+Enter), navigate to the Desktop, and manually press the NVDA shortcut. This restarts NVDA successfully and now everything works normally.
Is your experience the same? If so, I'm happy to contribute to this conversation. If not, my issue is distinct and should go elsewhere.

**Comment by Adriani90** [comment]
@yukie-nobuharu are you still having this issue with NVDA last alpha version?
You can download the last alpha version here:
https://www.nvaccess.org/files/nvda/snapshots/

**Comment by seanbudd** [comment]
@bhavyashah - which NVDA version are you reproducing this issue on?

**Comment by seanbudd** [comment]
Closing until more information is provided.
It seems that the issues raised by @bhavyashah might be better as a separate issue.
